heareval/tasks/util/luigi.py

```diff
@@ -63,6 +77,55 @@ def stage_number(self) -> int:
             raise ValueError("Unknown requires: %s" % self.requires())
 
 
+class DownloadCorpus(WorkTask):
+    """
+    Task for downloading a dataset to a specific filename
+    """
+
+    # URL to download dataset from
+    url = luigi.Parameter()
+    outfile = luigi.Parameter()
+
```

Comment:
Okay so you have to be really really careful here because this is a gotcha this is hard for us to detect automatically through code.

The checkpoint name is based upon self.name. If there are parameters and they are not used in self.name, then the checkpoints won't track multiple calls to this task. In this case, the task name is made unique by slugifying the in url.

Reply:
I mean maybe if we call it uid_including_parameters it might also be clearer?

Reply:
They have some notion of insignificant parameters, https://luigi.readthedocs.io/en/stable/parameters.html#insignificant-parameters but my understanding is that those are only applicable when running things from scratch, to determine they should not be part of the task signature. They won't fix the issue if you are re-running an incomplete pipeline.

Reply:
@khumairraj this is an important gotcha and curious also your thoughts here on how to protect against this

Reply:
We are currently saving the checkpoints inside each task subdirectory. This will prevent this problem to happen. What do you think?